packages/shared/src/schemas.ts

```diff
@@ -39,6 +42,104 @@ export const openSwapDepositChannelSchema = z
     expectedDepositAmount: amount,
   }));
 
+export const depositChannelIdRegex = z.string().regex(channelIdRegex);
+export const swapIdRegex = z.string().regex(/^\d+$/i);
+export const txHashRegex = z.string().regex(/^0x[a-f\d]+$/i);
+
+const swapStatusId = z.union([depositChannelIdRegex, swapIdRegex, txHashRegex]);
+
+export const getSwapStatusSchema = z.object({
+  id: swapStatusId,
+});
+
+export const swapState = z.enum([
+  'AWAITING_DEPOSIT',
+  'DEPOSIT_RECEIVED',
+  'SWAP_EXECUTED',
+  'EGRESS_SCHEDULED',
+  'BROADCAST_REQUESTED',
+  'BROADCAST_ABORTED',
+  'COMPLETE',
+]);
+
+const baseSwapResponseSchema = z.object({
+  srcChain: chainflipChain,
+  destChain: chainflipChain,
+  srcAsset: chainflipAsset,
+  destAsset: chainflipAsset,
+  destAddress: string.optional(),
+  depositAddress: string.optional(),
+  depositChannelCreatedAt: number.optional(),
+  expectedDepositAmount: string.optional(),
+  depositChannelExpiryBlock: z.bigint().optional().nullable(),
+  estimatedDepositChannelExpiryTime: number.optional(),
+  isDepositChannelExpired: z.boolean(),
+  ccmDepositReceivedBlockIndex: string.optional().nullable(),
+  ccmMetadata: z
+    .object({
+      gasBudget: string,
+      message: string.nullable(),
+    })
+    .optional()
+    .nullable(),
+});
+
+const awaitingState = baseSwapResponseSchema.extend({
+  state: swapState.refine((state) => state === swapState.Enum.AWAITING_DEPOSIT),
```

Comment:
if anyone has a better idea how to solve this. pls let me know. I can't use a zod literal here because trpc was complaining. It isn't ideal right now because we still need to do `if(x in prop)`

Reply:
what was the trpc error message?

Reply:
complaining about the state return types not matching 🤷